### PR TITLE
using an ordered map to iterate over the elements, instead of regular…

### PR DESCRIPTION
**Pull request by tomersolomon1**
Use ordered map to handle solve inconsistency issues.
In addition, move to Golang 1.18 (from 1.17). This required some minor change somewhere (some function was deprecated, moved to the new one).

Signed-off-by: tomersolomon1 <tomer.solomon1@gmail.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -9,6 +9,16 @@ require (
 	sigs.k8s.io/controller-tools v0.11.1
 )
 
+require (
```
Comment (roytman) [inline go.mod:12]: can you combine this and the next `require` blocks?

File: pkg/schemas/generator.go [hunk 2]
```diff
@@ -13,6 +13,8 @@ import (
 	"path/filepath"
 	"strings"
 
+	orderedmap "github.com/wk8/go-ordered-map/v2"
+
```
Comment (roytman) [inline pkg/schemas/generator.go:17]: pls remove this empty line

**Review by roytman — COMMENTED** [review]
Thanks for the PR, a couple of tiny comments

File: go.mod (hunk 1 above)
Comment (tomersolomon1) [reply go.mod:12]: I joined the two indirect blocks into a single one. Now there is one block for the indirect and one for the direct.

**Review by roytman — APPROVED** [review]

**Pull request merged**